internal/controller/stages/verification.go

```diff
@@ -32,41 +35,54 @@ func (r *reconciler) startVerification(
 
 	namespace := r.getAnalysisRunNamespace(stage)
 
-	// Check for existing AnalysisRun
-	analysisRuns := rollouts.AnalysisRunList{}
-	if err := r.listAnalysisRunsFn(
-		ctx,
-		&analysisRuns,
-		&client.ListOptions{
-			Namespace: namespace,
-			LabelSelector: labels.SelectorFromSet(
-				map[string]string{
-					kargoapi.StageLabelKey:   stage.Name,
-					kargoapi.FreightLabelKey: stage.Status.CurrentFreight.Name,
-				},
-			),
-		},
-	); err != nil {
-		return &kargoapi.VerificationInfo{
-			Phase: kargoapi.VerificationPhaseError,
-			Message: fmt.Errorf(
-				"error listing AnalysisRuns for Stage %q and Freight %q in namespace %q: %w",
-				stage.Name,
-				stage.Status.CurrentFreight.Name,
-				namespace,
-				err,
-			).Error(),
-		}
-	}
-	if len(analysisRuns.Items) > 0 {
-		logger.Debug("AnalysisRun already exists for Freight")
-		return &kargoapi.VerificationInfo{
-			Phase: kargoapi.VerificationPhase(analysisRuns.Items[0].Status.Phase),
-			AnalysisRun: &kargoapi.AnalysisRunReference{
-				Name:      analysisRuns.Items[0].Name,
-				Namespace: analysisRuns.Items[0].Namespace,
-				Phase:     string(analysisRuns.Items[0].Status.Phase),
+	// If the stage does not have a reverification annotation, check if there is
+	// an existing AnalysisRun for the Stage and Freight. If there is, return
+	// the status of this AnalysisRun.
+	if _, ok := stage.GetAnnotations()[kargoapi.AnnotationKeyReverify]; !ok {
```

Comment:
What happens here if we previously succeeded in restarting verification, but failed to clear the annotation?

It looks to me like that would result in a duplicate AnalysisRun.

Reply:
In that case (assuming the status update patch _did_ work), we would not end up here.

Reply:
🤦‍♂️